### ISSUE TITLE
pip-sync broken with editables

### ISSUE THREAD

**Issue by Chronial** [nvie/pip-tools#270]
pip-sync is broken with editables: It will switch between installing and uninstalling them in consecutive installations:

```
vagrant@v-ston ~/t/envtest> cat requirements.txt
-e git+https://github.com/Xof/django-pglocks.git@f5df93e#egg=django-pglocks

vagrant@v-ston ~/t/envtest> virtualenv -p python3 .
Running virtualenv with interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in ./bin/python3
Also creating executable in ./bin/python
Installing setuptools, pip, wheel...done.

vagrant@v-ston ~/t/envtest> . bin/activate.fish

(envtest)vagrant@v-ston ~/t/envtest> pip install pip-tools
Collecting pip-tools
  Using cached pip_tools-1.3.0-py2.py3-none-any.whl
Collecting first (from pip-tools)
  Using cached first-2.0.1-py2.py3-none-any.whl
Collecting six (from pip-tools)
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting click>=6 (from pip-tools)
  Using cached click-6.2-py2.py3-none-any.whl
Installing collected packages: first, six, click, pip-tools
Successfully installed click-6.2 first-2.0.1 pip-tools-1.3.0 six-1.10.0

(envtest)vagrant@v-ston ~/t/envtest> pip-sync
Collecting django-pglocks from git+https://github.com/Xof/django-pglocks.git@f5df93e#egg=django-pglocks
  Cloning https://github.com/Xof/django-pglocks.git (to f5df93e) to /tmp/pip-build-ete6jfh2/django-pglocks
  Could not find a tag or branch 'f5df93e', assuming commit.
Installing collected packages: django-pglocks
  Running setup.py install for django-pglocks
Successfully installed django-pglocks-1.0.2

(envtest)vagrant@v-ston ~/t/envtest> pip-sync
Uninstalling django-pglocks-1.0.2:
  Successfully uninstalled django-pglocks-1.0.2
Requirement already satisfied (use --upgrade to upgrade): django-pglocks from git+https://github.com/Xof/django-pglocks.git@f5df93e#egg=django-pglocks in ./lib/python3.4/site
-packages

(envtest)vagrant@v-ston ~/t/envtest> python
Python 3.4.3 (default, Oct 14 2015, 20:28:29)
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import django_pglocks
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named 'django_pglocks'
```

**Comment by skolsuper** [comment]
+1, I have to run it twice every time since upgrading to 1.3 (I was on 1.1.6 before).

**Comment by nvie** [comment]
This has been addressed in 1.4.1. It was a weird bug, but fixed in 8f64896e11a1d34a27b4f4c5d12bed04336a7c75. Thanks for finding/reporting!
